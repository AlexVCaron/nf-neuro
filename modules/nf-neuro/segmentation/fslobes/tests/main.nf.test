nextflow_process {

    name "Test Process SEGMENTATION_FSLOBES"
    script "../main.nf"
    process "SEGMENTATION_FSLOBES"

    tag "modules"
    tag "modules_nfcore"
    tag "segmentation"
    tag "segmentation/fslobes"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    config "./nextflow.config"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "freesurfer_no_surf.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("segmentation - fs_lobes") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/", checkIfExists: true)
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("segmentation - fs_lobes - stub-run") {
        tag "stub"
        options "-stub-run"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/", checkIfExists: true)
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
