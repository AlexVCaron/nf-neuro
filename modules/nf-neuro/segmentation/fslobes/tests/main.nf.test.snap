{
    "segmentation - fs_lobes": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__atlas_lobes.nii.gz:md5,8fa99f47b9889f173b552afc889409c6"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__atlas_lobes_dilate.nii.gz:md5,bca87675be74667711b050c35c72172e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,63a1a6726b66177711ba49f750a62b9b"
                ],
                "labels": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__atlas_lobes.nii.gz:md5,8fa99f47b9889f173b552afc889409c6"
                    ]
                ],
                "labels_dilate": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__atlas_lobes_dilate.nii.gz:md5,bca87675be74667711b050c35c72172e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,63a1a6726b66177711ba49f750a62b9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-17T18:31:43.896509234"
    },
    "segmentation - fs_lobes - stub-run": {
        "content": [
            [
                "versions.yml:md5,63a1a6726b66177711ba49f750a62b9b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-17T18:29:57.530200468"
    }
}